HYDRATOR-260 - Pasting a list of field names into Hydrator schema should populate multiple rows

sushil-car opened the pull request:
https://issues.cask.co/browse/HYDRATOR-260

sushil-car committed:
HYDRATOR-260 - Pasting a list of field names into Hydrator schema

russorat commented:
@ajainarayanan could you have someone from the UI team check this?

abhinavbansal246 commented on cdap-ui/app/directives/complex-schema/complex-schema.js:
```diff
@@ -68,6 +68,32 @@ function ComplexSchemaController (avsc, SCHEMA_TYPES, $scope, uuid, $timeout, Sc
+  vm.pasteFields = (event, field) => {
+
+    let pastedData = event.clipboardData.getData('text/plain');
+    let pastedDataArr = pastedData.replace(/[\n\r\t ]/g, '$#$').split('$#$');
```
Please also add "," and "|" in the list of splitters. It would be really nice if we can just paste the headers of the csv file and generate schema from that

ajainarayanan commented:
@sushiladokar It would be nice if there is error handling. Right now if I import a huge schema it takes a while (with UI being unresponsive) .

Also I was not sure what I can copy paste. It would nice if we had some UI/UX way of saying to the user that they can copy paste multiple strings separated by tab or newline @russorat Did you get a chance to talk to Lea about it? Might be very useful if we could surface this feature in a more visible way.

ajainarayanan commented:
Aside from the above comment codewise 👍 LGTM

russorat commented:
@ajainarayanan does restricting the UI to 300 fields satisfy things?

ajainarayanan commented:
@russorat So the aim was not to have restrict the number of rows as we anyways allow the user to import any number of rows for the schema.

The idea was to show a loading icon while processing huge schemas and indicate any error if there are any. Just like import feature. Right now the following happens,
- If I copy-paste huge schema in the first row the UI becomes unresponsive
- Initially I copy pasted the json schema instead of the raw text which resulted in a broken state in UI. May be we should do error validations before updating the schema based on copy paste.

We shouldn't be restricting the number of rows in the schema being imported as it is not the same as how we allow users to import the same.

prinam commented:
So just to be clear, action items for this PR:

- It should show a loading icon if a large schema is imported
- It should show appropriate error message if invalid schema is imported

sushil-car commented:
Addressing the review comments
I  have used the same functionality as used when importing a schema. But as import schema overrides the existing list and populates new rows, it will also use new records, when pasted. Let me know if this is correct behaviour.

elfenheart commented:
@sushiladokar the behavior should be that the pasted data should append to the existing schema. Importing schema is going to overwrite, we shouldn't overwrite the schema.

russorat commented:
@elfenheart @ajainarayanan could you take a second look?

elfenheart commented:
👍  LGTM. Please squash commits

sushil-car commented:
Squashed the commits
